File: Scripts/laptop/file-input-output/r/file-input-output.r

r = 1

executeTask <- function(i) {
	fileLocation = paste("../test_directory/", i, sep = "")
	src <- file(fileLocation, "r")
	dest <- file("output.txt", "w")

	fc <- readLines(src, -1)
	writeLines(fc, dest)
	close(src); close(dest)
	return(i)
}

for ( i in 1:10001 ) {
	r = executeTask(i)
}

File: Scripts/laptop/file-input-output/java/FileInputOutput.java
import java.io.*;
public class FileInputOutput {

  private static volatile int r; 
 
  public static int executeTask(int i)  throws IOException{

    BufferedReader br = new BufferedReader(new FileReader("../test_directory/"+i));
    BufferedWriter bw = new BufferedWriter(new FileWriter("output.txt"));
    String line;
    while ((line = br.readLine()) != null) {
      bw.write(line);
      bw.newLine();
    }
    br.close();
    bw.close();
	return i + 1;
  }

  public static void main (String[] args) throws IOException {

    	for ( int i = 1; i <= 10000; ++i) {
		r = executeTask(i);
	 }
  }
}
